odoo/orm/utils.py

```diff
@@ -145,3 +178,29 @@ def __reversed__(self):
         for id_ in reversed(self.ids):
             if id_ := id_ or id_.origin:
                 yield id_
+
+
+class ConcatIds(Reversible):
+    """ A reversible iterable returning the union of collections of ``ids``. """
+    __slots__ = ['_iterables']
+
+    def __init__(self, iterables):
+        # flatten out iterables to avoid a recursion error when iterating
+        result = []
+        for it in iterables:
+            result.extend(it._iterables) if isinstance(it, ConcatIds) else result.append(it)
+        self._iterables = tuple(result)
+
+    def __iter__(self):
+        for iterable in self._iterables:
+            yield from iterable
+
+    def __reversed__(self):
+        for iterable in reversed(self._iterables):
+            yield from reversed(iterable)
+
+
+class Prefetch:
+    relational = PrefetchRelational
+    origin = OriginIds
+    union = ConcatIds
```

Comment:
```suggestion
    concat = ConcatIds
```
More sense no deduplication